public/app/features/browse-dashboards/components/CheckboxHeaderCell.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+import { Checkbox } from '@grafana/ui';
+
+import { DashboardTreeHeaderProps, SelectionState } from '../types';
+
+export default function CheckboxHeaderCell({ isSelected, onAllSelectionChange }: DashboardTreeHeaderProps) {
+  const state = isSelected?.('$all') ?? SelectionState.Unselected;
```

Comment:
So this checkbox is not reactive to selecting "children" at all? For instance if I select ALL folders/dashboards, this doesn't get ticket. But at the same time if I click the checkbox it selects all, but then if I deselect one folder, it is still ticked. Did we want to add an indeterminate state here as well or what should be the functionality of this checkbox?

Reply:
> For instance if I select ALL folders/dashboards, this doesn't get ticket.

Hmm. It should. This sounds like a bug :)